### DISCUSSION TITLE
FAT32 32K for 2GB card?

### DISCUSSION THREAD

**Discussion by salzrat** [jomjol/AI-on-the-edge-device#2969] [Q&A]
The docs say you should use FAT32 32K as format for the SD card. However, for a 2GB SD-card, 32K clusters are not possible. 
Are the 32K cluster just a recommendation or required? I can use regular FAT16 to get 32K clusters. Or just stick with FAT32, 4K clusteres?

**Comment by SybexX** [comment]
It doesn't really matter, the main thing is Fat32.
I use 115MB sd and format it in Fat32 and leave the rest as standard

**Reply by salzrat** [reply]
What is the advantage of FAT32 over FAT16 in small sticks?
For larger SD-cards, does the cluster size really matter? Otherwise, maybe the docs should be changes, it is quite confusing...

**Comment by SybexX** [comment]
The implementation/support is from Espressif, I don't know whether FAT16 is supported.
Since the ESP32 is a 32-bit CPU and FAT32 is also 32-bit, I assume that the speed is reduced with Fat16 and it requires more memory.
https://www.easeus.de/partitionieren-tipps/fat-dateisystemformat.html

**Comment by salzrat** [comment]
This doesn't really have anything to do with CPU bits. FAT32 just uses 32bit for indexing clusters, so it can use more clusters and thus address larger drives. 2GB can still be handled with FAT16. The difference is just the cluster size - FAT16 requires 32K clusters for 2GB, whereas FAT32 can use 4KB clusters, leading to less wastage. However, for for this purpose, this is mostly irrelevant as we are only using a small fraction of the space anyway...
and yes, FAT16 is supported, I tried it...

**Comment by SybexX** [comment]
I didn't bother withI didn't bother with it that much, I assumed that a cluster at Fat32 is also divided up 32-bit words.
you can find more information about it there: https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/storage/fatfsgen.html
